lib/graphql/stitching/composer/resolver_config.rb

```diff
@@ -31,14 +31,14 @@ def extract_federation_entities(schema, location)
               next unless directive.graphql_name == "key"
 
               key = directive.arguments.keyword_arguments.fetch(:fields).strip
-              raise ComposerError, "Composite federation keys are not supported." unless /^\w+$/.match?(key)
+              raise CompositionError, "Composite federation keys are not supported." unless /^\w+$/.match?(key)
```

Comment:
This restriction could be removed, though we'd need to parse the key and spread all root fields into the argument format.